### PR TITLE
fixed some errors on index.js in back folder

### PR DESCRIPTION
**Pull request by Brolyus**

### CODE REVIEW

File: back/index.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ app.route('/register').post(async (request, response) => {
 		const user = { email: request.body.email, password: hashedPassword }
 
 		connection.query(
-			'SELECT email FROM account WHERE email = ?',
+			`SELECT email FROM account WHERE email = ${user.email}`,
```
Comment (rocambille) [inline back/index.js]: no: this makes you vulnerable to sql injection. try:

    		connection.query(
			'SELECT email FROM account WHERE email = ?',
                        [user.email],
                        ...

File: back/index.js [hunk 2]
```diff
@@ -113,46 +113,17 @@ app.route('/login').post((request, response) => {
...
 		jwt.verify(param, secret, (err, authData) => {
 			const userEmail = authData.iss
-			console.log(authData)
 			if (err) {
 				response.sendStatus(401)
 			} else {
 				connection.query(
-					`SELECT id, picture, nickname, biography, ville FROM profile WHERE id = '${idProfile}'`,
+					`SELECT profile.id FROM profile INNER JOIN account ON email = '${userEmail}' WHERE profile.account_id  = account.id`,
 					[idProfile],
```
Comment (rocambille) [inline back/index.js]: same here: prefer `?` markers

File: back/index.js [hunk 3]
```diff
@@ -185,6 +156,34 @@ app.route('/profiles/:id')
...
+	.get((request, response) => {
+		const param = request.query.token
+		const idProfile = request.params.id
+		jwt.verify(param, secret, (err, authData) => {
+			if (err) {
+				response.sendStatus(401)
+			} else {
+				connection.query(
+					`SELECT id, picture, nickname, biography, ville FROM profile WHERE id = '${idProfile}'`,
+					[idProfile],
```
Comment (rocambille) [inline back/index.js]: same here

**Review by rocambille — CHANGES_REQUESTED** [review]

**Review by Brolyus — COMMENTED** [review]
Corrected the sql vulnerabilites

**Pull request merged**